Automattic/jetpack
# Decouple subscription block from Jetpack's subscription widget shortcode

Fixes #26866.

#### Changes proposed in this Pull Request:

This is an internal PR which solves the issue of the subscription block being closely dependent on Jetpack Subscription Widget for rendering the form - resulting in less flexibility and more difficult development process with the need of taking both into account.  
In order to accomplish this, the subscribe block is now a dynamic block, employing `render_block()` to render it from the backend/PHP instead of relying on the block editor's `save()` method.

While there's still _some_ dependency on the `Jetpack_Form_Widget` class for form handling purposes, the view itself is now completely independent from the shortcode/widget.

I think this is a good first step, and now that it's done it'll probably allow for eventual optimization and simplification of the block code, but I'd argue that's outside of the scope of this PR itself.  

![Screenshot 2022-10-19 at 22 28 18](https://user-images.githubusercontent.com/8056203/196801934-13c8f3b8-1d33-4de5-9586-c3f5a78e3e41.png)

#### Other information:


#### Jetpack product discussion



N/A

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






This PR is meant to be an internal change only, hence the best way of testing is side by side on two separate sites with the current and the new implementation and compare.

- Make sure subscriptions are enabled on your site (Jetpack->Discussion->Subscriptions). You will also need the connection to Jetpack/WP.com to be available during the testing.
- Create or edit a post and add a Subscription block.
- When you publish, the subscription block display normally on the page.
- You should be able to submit the form using a correct email address, after which a success message will be shown and you'll receive a confirmation email.
- When you submit an invalid email, you should see an error.
- Try the various style setting of the block, including displaying the button on a new line or adding the subscriber count. Everything should work as it does in the existing version.
